Automattic/jetpack
# Story block: remove wp-components frontend dependency

The story block frontend currently enqueues `wp-components`, which is an absolutely massive 200KB (compressed) dependency, and which comes with other large dependencies of its own.

It does this for two different reasons:
- To reuse some SVG-related components. While this is ripe for a deeper refactor, an immediate improvement we can apply is to use `wp-primitives` instead; it has the exact same components (`wp-components` reexports them), and `wp-primitives` has a massively smaller footprint.
- To reuse the `<Button>` component. None of the usage seems to justify it, so I simply replaced it with `<button>`.

Note: I can't find a particular team that is currently responsible for this block, so I'm assigning to @Automattic/jetpack-crew. Please feel free to reassign!

This PR should save ~250KB compressed (!!) on pages with story blocks.

## Proposed changes:
* Story block: remove dependency on `wp-components` SVG-related components by replacing it with a dependency on the `wp-primitives` equivalents
* Story block: remove dependency on `wp-components` `<Button>` and replace with plain `<button>`

### Other information:

- N/A Have you written new tests for your changes, if applicable?

## Jetpack product discussion
None, this is a simple performance improvement with no functionality changes.

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
- Create a page with a story block
- Load the page and ensure that `wp-components` is not loaded
- Smoke-test the story block and ensure it works correctly
